cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -81,26 +87,40 @@ import           Cardano.Api.ProtocolParameters
 import           Cardano.Api.TxBody
 import           Cardano.Api.Value
 
+import           Data.Word (Word64)
 
 -- ----------------------------------------------------------------------------
 -- Queries
 --
 
 data QueryInMode mode result where
+  QueryCurrentEra
+    :: ConsensusModeIsMultiEra mode
+    -> QueryInMode mode AnyCardanoEra
 
-     QueryCurrentEra :: ConsensusModeIsMultiEra mode
-                     -> QueryInMode mode AnyCardanoEra
+  QueryInEra
+    :: EraInMode era mode
+    -> QueryInEra era result
+    -> QueryInMode mode (Either EraMismatch result)
 
-     QueryInEra      :: EraInMode era mode
-                     -> QueryInEra era result
-                     -> QueryInMode mode (Either EraMismatch result)
+  QueryEraHistory
+    :: ConsensusModeIsMultiEra mode
+    -> QueryInMode mode (EraHistory mode)
 
 --TODO: add support for these
 --     QueryEraStart   :: ConsensusModeIsMultiEra mode
 --                     -> EraInMode era mode
 --                     -> QueryInMode mode (Maybe EraStart)
 
---     QueryEraHistory :: QueryInMode mode EraHistory
+data EraHistory mode where
+  EraHistory
+    :: ConsensusBlockForMode mode ~ Consensus.HardForkBlock xs
+    => ConsensusMode mode
+    -> History.Interpreter xs
+    -> EraHistory mode
+
+slotToEpoch :: SlotNo -> EraHistory mode -> Either Qry.PastHorizonException (EpochNo, Word64, Word64)
```

Comment:
What are the two `Word64`s? A comment here would help.

Reply:
Added type aliases to help with that.

Reply:
Where are the type aliases? Would it not be better the be a `newtype` ?